src/components/ListItem.jsx

```diff
@@ -35,7 +35,7 @@ export function ListItem({ name, listPath, itemId, dateLastPurchased }) {
 	useEffect(() => {
 		const handleUpdateItem = async () => {
 			if (isChecked) {
-				await updateItem(listPath, itemId);
+				await updateItem(listPath, itemId, dateLastPurchased);
```

Comment:
I am getting an error:
 `Uncaught (in promise) TypeError: Cannot read properties of null (reading 'toDate')
    at updateItem (firebase.js:197:21)
    at async handleUpdateItem (ListItem.jsx:38:5)`. 
    
 It looks like you need to add `dateLastPurchased` to the dependency array in this `useEffect` and check that it exists before running `updateItem`. That property will be null for an item that hasn't been purchased before so you can just pass in a new date in that case.

Reply:
You don't want a new date there, however. The answer is a little obscure

https://github.com/the-collab-lab/shopping-list-utils/blob/main/src/calculateEstimate/calculateEstimate.ts

> daysSinceLastTransaction: number, // The number of days since the item was added to the list or last purchased

If dateLastPurchased is null what should you use instead?

Reply:
Thank you. Becuase all my items has a dateLastPurchased before, so i didn't notice this.

Our solution: In firebase updateItem, we decleared a new variable `const effectiveDateLastPurchased = dateLastPurchased
		? dateLastPurchased.toDate()
		: new Date();` and do the daysSinceLastPurchase calculation and call updateDoc with `effectiveDateLastPurchased `

Reply:
See my previous comment. You don't want a new date there. You want the date that the item was created if dateLastPurchased is null